resources/prosody-plugins/mod_filter_iq_jibri.lua

```diff
@@ -10,12 +15,30 @@ module:hook("pre-iq/full", function(event)
         if jibri then
             local session = event.origin;
             local token = session.auth_token;
+            local feature = jibri.attr.recording_mode == 'file' and 'recording' or 'livestreaming';
+            local is_allowed = is_feature_allowed(session.jitsi_meet_context_features, feature);
+
+            -- if current user is not allowed, but was granted moderation by a user
+            -- that is allowed by its features we want to allow it
+            local is_granting_allowed = false;
+            if session.granted_jitsi_meet_context_features then
+                is_granting_allowed = is_feature_allowed(session.granted_jitsi_meet_context_features, feature);
+            end
 
             if jibri.attr.action == 'start' then
-                if token == nil
-                    or not is_feature_allowed(session.jitsi_meet_context_features,
-                    (jibri.attr.recording_mode == 'file' and 'recording' or 'livestreaming')
-                ) then
+                if token == nil or not (is_allowed or is_granting_allowed)
+                then
+                    if not session.jitsi_meet_context_features and not session.granted_jitsi_meet_context_features then
+                        -- we need to check for moderator rights
+                        -- when there are no features and the occupant is moderator we allow recording
```

Comment:
Why don't we extract the set of features in a central place? I think it would be less error prone. For example, we can do it at token verification time. We can read the token and set the features accordingly. If the moderator flag is raised and no features are specified we can populate the defaults. That would also be the place to set features to an empty set when no token is present. Then, everywhere else we can assume that the set of features is set accordingly, and we don't have to know the details.

This whole section would simplify to:
```
local is_allowed = ...
if ~is_allowed then
  module:log(...)
  session.send(st.error_reply(stanza, "auth", "forbidden"));
end  
```

Reply:
We do not have a common rule for defining moderator, and ther rules we use in deploments are outside of this project. 
The extraction time of features is before we know whether it is a moderator.

Reply:
I see. Then can we extract a utility function like `is_feature_allowed(features, granted_features, isModerator)? So that all that logic is consistent and in one place?

Reply:
I think I addressed that now.

Reply:
We lost some small optimizations cause before we were looking up the room to search for the occupant to check the role only when needed.